flink-cdc-connect/flink-cdc-source-connectors/flink-cdc-base/src/main/java/org/apache/flink/cdc/connectors/base/source/assigner/state/PendingSplitsStateSerializer.java

```diff
@@ -390,7 +390,7 @@ private void writeTableIds(Collection<TableId> tableIds, DataOutputSerializer ou
         for (TableId tableId : tableIds) {
             boolean useCatalogBeforeSchema = SerializerUtils.shouldUseCatalogBeforeSchema(tableId);
             out.writeBoolean(useCatalogBeforeSchema);
-            out.writeUTF(tableId.toString());
+            out.writeUTF(tableId.toDoubleQuotedString());
```

Comment:
Could you fix org.apache.flink.cdc.connectors.mysql.source.assigners.state.PendingSplitsStateSerializer as well ?